src/control_flow/mod.rs

```diff
@@ -86,12 +101,17 @@ struct Scope<'a> {
   found_break: Option<Option<Id>>,
   found_continue: bool,
 }
-#[derive(Debug, Copy, Clone)]
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
 enum Done {
   /// Return, Throw, or infinite loop
   Forced,
-  // Break or continue
+  /// Break or continue
   Break,
+  /// Pass through a block, like a function's block statement which ends without returning a value
+  /// or throwing an exception. Note that a node marked as `Done::Pass` won't prevent further execution, unlike
+  /// `Done::Forced` and `Done::Break`.
+  Pass,
```

Comment:
It's a nitpicking, but this name seems bit confusing. I used `Option<Done>` because it's clear that it will continue execution if `done` is `None`.

Reply:
I misread the comment. Please ignore the comment about `Option<Done>`.

But I think `Done::Pass` is confusing anyway. 
How about `Fallthrough`? It seems like a commonly used alternative.

Reply:
Maybe 

 - `Done` => `End`
 - `Pass` => `Continue`
 - `mark_as_done` => `mark_as_end`

would be better